doc/directives.rst

```diff
@@ -0,0 +1,185 @@
+.. _directives:
+
+Directives
+==========
+
+Hawkmoth provides several new directives for incorporating :ref:`documentation
+comments <syntax>` from C source files into the reStructuredText document. There
+are three main types of directives, for incorporating documentation from entire
+files, for single objects, and for composite objects optionally with members.
+
+Source Files
+------------
+
+The :rst:dir:`c:autodoc` directive simply includes all the documentation
+comments from any number of files. This is the most basic and quickest way to
+generate documentation, but offers no control over what gets included.
+
+.. rst:directive:: .. c:autodoc:: filename-pattern [...]
```

Comment:
Sorry to "reopen" the review, but while testing other things, I noticed these warnings the 1st time I ran `make html`. However, I cannot reproduce it any more, no matter how well I clean up the working directory. Not sure if it originated in some race condition... Have you seen these before?

Either way, I wonder if we shouldn't instead put all the documentation in the Python code and incorporate it here through Python's own autodoc.

```
hawkmoth/doc/directives.rst:18: WARNING: duplicate description of directive c:autodoc, other instance in extension
hawkmoth/doc/directives.rst:24: WARNING: duplicate description of directive:option c:autodoc:transform, other instance in extension   
hawkmoth/doc/directives.rst:31: WARNING: duplicate description of directive:option c:autodoc:compat, other instance in extension
hawkmoth/doc/directives.rst:43: WARNING: duplicate description of directive:option c:autodoc:clang, other instance in extension
hawkmoth/doc/directives.rst:68: WARNING: duplicate description of directive c:autovar, other instance in extension
hawkmoth/doc/directives.rst:73: WARNING: duplicate description of directive:option c:autovar:file, other instance in extension        
hawkmoth/doc/directives.rst:87: WARNING: duplicate description of directive c:autotype, other instance in extension
hawkmoth/doc/directives.rst:96: WARNING: duplicate description of directive c:automacro, other instance in extension
hawkmoth/doc/directives.rst:105: WARNING: duplicate description of directive c:autofunction, other instance in extension
hawkmoth/doc/directives.rst:127: WARNING: duplicate description of directive c:autostruct, other instance in extension
hawkmoth/doc/directives.rst:133: WARNING: duplicate description of directive:option c:autostruct:members, other instance in extension
hawkmoth/doc/directives.rst:162: WARNING: duplicate description of directive c:autounion, other instance in extension
hawkmoth/doc/directives.rst:172: WARNING: duplicate description of directive c:autoenum, other instance in extension                  
```

Reply:
I think it has to be because the previous version had the documentation in `extension.rst` and now I moved all the directive documentation to a separate `directives.rst`. Maybe Sphinx got confused with what it had cached for an earlier run?

Reply:
Ah good call, just checked and that was it in fact.